### ISSUE TITLE
Razor .dll

### ISSUE THREAD

**Issue by OpansKeyrist** [nipkownix/re4_tweaks#116]
hello, I have a problem combining Raz0r .dll and RE4_Tweaks. I've renamed dinput8.dll(RE4_Tweaks) to winmm.dll
but crashes when opening inventory. is there something wrong?

**Comment by nipkownix** [comment]
Try turning RaiseInventoryAlloc off. Raz0r's dll does something that conflicts with our option, and there's no way to turn of his.

**Comment by KsaNL** [comment]
He's dll has a shelling make IDA can't work 
I took OD dump a list from Raz0r's dll completely, but just frist in game check. 
Including cracking pos for that which can play without steam, you need i'll provide. ![url](https://ibb.co/0KjxXwQ)

**Comment by Mister-Curious** [comment]
@nipkownix @emoose So I had my engineer friend take the decrypted contents of Raz0r's DLL (by reading them from memory once they are loaded), and place them in a file. Essentially the DLL is encrypted, but it still needs to be decrypted into memory. So he took this from memory and placed it into [THIS FILE](https://drive.google.com/file/d/1fF2CCtdNsqR-nFYMKWdIMJ5YzGNbX8kg/view?usp=sharing). 

Open this file in a hex editor and Start looking for "CC CC CC CC CC", these are where the functions are. My engineer friend  tried turning this assembly into a dll, but the game wouldn't run. Despite that all the functions in memory are there in that file. 

I am providing this with hopes that perhaps we can rebuild Raz0r's DLL without the encryption in order to remove/fix any of the functions that conflict with the RE4 Tweaks DLL. I recently asked someone on this site if it would be possible to create code out of raw assembly, and I think it was @emoose who said it would be possible. Please let me know your thoughts on this, and if there is any assistance I, or my engineer friend can provide. 

Cheers

**Comment by KsaNL** [comment]
> Open this file in a hex editor and Start looking for "CC CC CC CC CC", these are where the functions are. My engineer friend tried turning this assembly into a dll, but the game wouldn't run. Despite that, all the functions in memory are there in that file.
Frist Best wishes for the great work.
Except game can load the dll and work directly, just dump it from the memory help hardly.
We should work without Raz0r's dll, seems he did't want others extract function from his dll.
And we have the dump cpu info, so that it's easy to find the crash pos and then try to fix or just banned Raz0r's trainner.
All of my opinions.

**Comment by Mister-Curious** [comment]
@KsaNL I don't think it matters what was the intended use of Raz0r's dll. I think its more about creating a working environment that provides the most functionality for all.

**Comment by qingsheng8848** [comment]
We don't need to crack razor's dll, we can completely recreate the functions he has on the dll

**Comment by qingsheng8848** [comment]
We don't need to crack his DLL to make it work in piracy.We are all likely to implement his DLL function, I have found the code of the game loaded stage .udas file, and I have made a DLL, which can load all ITM models.

**Comment by Mister-Curious** [comment]
@qingsheng8848 of course your work has been magnificent, and I look forward to seeing more progress from you. At this moment though there is much functionality that has not been paralleled by any other coders yet,  like the ability to side-chain files, and to pay movies etc. 

If you think you can duplicate all the functionality of the Companion DLL, and do so in an open source format so that the community can observe and edit the code as needed, I welcome that 100%. I just thought it might be worth providing the assembly code of the Companion DLL to at least observe what its doing when conflicts are happening with the RE4 Tweaks DLL

**Comment by OpansKeyrist** [comment]
@nipkownix @KsaNL @Mister-Curious @qingsheng8848 
apart from what @Mister-Curious  said .dll raz0r works to load very large polys i.e. 55k which is initially only 20k, it can also store a lot of vertex data in it, which we call memory reserve. This is very important to me. Hopefully there is a middle way to unite the two or another way

**Comment by qingsheng8848** [comment]
> 我

https://residentevilmodding.boards.net/thread/16163/new-dll-publishing
Loading more polygons is not a problem, the key is to give the bones more delicate weights
